app/digests/tests/test_digest_api.py

```diff
@@ -134,3 +135,43 @@ def test_can_filter_on_digest_stage(stage: str,
     assert response.data['total'] == 1
     assert response.data['items'][0]['stage'] == stage
 
+@pytest.mark.django_db
+def test_is_paginated(client: Client, multiple_published_digests):
+    per_page = 3
+    def _count_items(page):
+        response = client.get(f'{DIGESTS_URL}?per-page={per_page}&page={page}')
+        assert response.status_code == 200
+        assert response.data['total'] == len(multiple_published_digests)
+        items = len(response.data['items'])
+        assert items <= per_page
+        return items
+
+    last_page = 4
+    total = sum([_count_items(page) for page in range(1, last_page)]) + _count_items(page=last_page)
+
+    assert total == len(multiple_published_digests)
+
+@pytest.mark.django_db
+def test_invalid_page(client: Client):
+    response = client.get(f'{DIGESTS_URL}?page=0')
+    assert response.status_code == 404
+
+@pytest.mark.django_db
+def test_invalid_per_page_is_ignored(client: Client, multiple_published_digests):
+    response = client.get(f'{DIGESTS_URL}?per-page=0')
+    assert response.status_code == 200
+    assert len(response.data['items']) == len(multiple_published_digests)
+
+@pytest.mark.django_db
+def test_page_too_large(client: Client, multiple_published_digests):
+    all_digests = len(multiple_published_digests)
+    response = client.get(f'{DIGESTS_URL}?per-page={all_digests}&page=2')
+    assert response.status_code == 404
+
+@patch('digests.pagination.DigestPagination.max_page_size', 5)
```

Comment:
Erm...

Reply:
Better than inserting 101 items in the database for a single test?

Reply:
Shouldn't need any items...

Reply:
We can probably pre-validate some of these in the `api` module, to cater for various 400 errors.